### PR TITLE
04 api/articles/article

### PR DESCRIPTION
**Pull request by RKHarrison**
Hey NC!

I've added the GET /api/articles/:article_id endpoint:
-adds happy path testing and functionality
-adds 404 not found testing and functionality for valid-but-non-existent id requests
-adds 400 bad request testing and functionality for invalid id schema syntax
-updates endpoints.json

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -50,4 +50,34 @@ describe('GET /api', () => {
...
+
+describe('GET /api/articles/:article_id', () => {
+    it('200: responds with an object matching the requested id, with correct properties', () => {
+        return request(app)
+        .get('/api/articles/1')
+        .expect(200)
+        .then(({body:article}) => {
+            const expectedResult = endpointsJsonFile["GET /api/articles/:article_id"].exampleResponse.article
+            expect(article).toMatchObject(expectedResult)
+        })
```
Comment (nc-linda-guester) [inline __tests__/app.test.js:63]: I wouldn't necessary expect to see this - I would expect a check to see whether the article you got back has the correct article_id (and other necessary properties)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -50,4 +50,34 @@ describe('GET /api', () => {
...
+    it("404: responds with 'Not Found' when given valid but non-existing id", () => {
+      return request(app)
+      .get('/api/articles/987654321')
+      .expect(404)
+      .then(({body}) =>{
+        const errorMsg = body.msg
+        expect(errorMsg).toBe("Not Found")
+      })
+    })
+    it("400: responds with 'Bad Request' when failing schema validation", () => {
```
Comment (nc-linda-guester) [inline __tests__/app.test.js:74]: Well done on the testing!

**Review by nc-linda-guester — COMMENTED** [review]
All in all, really well done! Nice implementation of the error handling, might be good to send the response back on a specific fitting key but apart from that, well done!

**Comment by RKHarrison** [comment]
Thanks for the feedback Linda, appreciate it. I will change my tests to
align with your suggestions in terms of what it checks for on the happy
path.

What do you mean by 'send the response back on a specific fitting key'? Do
you mean a custom error message or a more appropriate variable name? But
confused there.

Thanks again for your help 🐱

**Pull request merged**